### PR TITLE
Fix `Arbitrary` instance for `Var`

### PR DESCRIPTION
**Pull request by Gabriel439**
... by disallowing backtick characters in the label

Fixes https://github.com/dhall-lang/dhall-haskell/issues/1941

### CODE REVIEW

**Comment by Gabriel439** [comment]
@sjakobi: Yeah, I don't know a good way to fix that short of newtyping the `Text` field of the `Field` constructor.  The issue is that the `generic-random` code for deriving the `Arbitrary` instance for `Expr` relies on the `Arbitrary` instance for `Text` when generating the `Field` constructor

**Comment by sjakobi** [comment]
I wonder whether we could use `generic-random`'s `ConstrGen` feature to tweak the `Field`s. See the example at the bottom of this page: http://hackage.haskell.org/package/generic-random-1.3.0.1/docs/Generic-Random-Tutorial.html

File: dhall/tests/Dhall/Test/QuickCheck.hs [hunk 1]
```diff
@@ -275,17 +285,31 @@ instance (Arbitrary s, Arbitrary a) => Arbitrary (Expr s a) where
...
+            :+ ConstrGen "Pi" 0 Text
+            :+ ConstrGen "Field" 1 Text
+            :+ ConstrGen "Project" 1 (Either (Set Text) (Expr s a))
             :+ ()
         customGens =
                integer
                -- 'Lam's and 'Pi's are encoded differently when the binding is
                -- the special string "_", so we generate some of these strings
                -- to improve test coverage for these code paths.
             :+ Test.QuickCheck.oneof [pure "_", arbitrary]
```
Comment (sjakobi) [inline dhall/tests/Dhall/Test/QuickCheck.hs:300]: It seems to me that this generator might override the `ConstrGen` generators below. At least I can still reproduce https://github.com/dhall-lang/dhall-haskell/issues/1941 when it's there, and removing it seems to fix #1941.

(CC @Lysxia)

Comment (Lysxia) [reply dhall/tests/Dhall/Test/QuickCheck.hs:300]: Yeah, the first match wins. generic-random's doc could be clearer on that point.

Comment (sjakobi) [reply dhall/tests/Dhall/Test/QuickCheck.hs:300]: Thanks, @Lysxia! :)

In that case it probably makes sense to move these field-type-specific generators below the `ConstrGen` generators.

**Pull request merged**